File: src/lib/langchain/prompts.ts
import { ChatPromptTemplate } from "@langchain/core/prompts";

const chat_system_template = `
            "You are an assistant for question-answering tasks. "
            "Use the following pieces of retrieved context to answer the question "
            "If you don't know the answer, say that you don't know."
            "Use three sentences maximum and keep the answer concise."
            "\n\n"
            "{context}"
            
            `;

const chunks_system_template = `
            You are an expert assistant with expertise in summarizing research papers,
            Summarize the following text as detailed as possible \n\n
            Text: \n {text}\n
            Summary: 

            `;

const final_system_combine_template = `
            You are an expert assistant with expertise in summarizing research papers,
            Provide a complete summary of the entire Research Paper without missing any topic, in a simple language,
            Each topic after their explanation must have a simple example to explain further. \n\n
            Summary must be in markdown format. \n\n
            Text: \n {text}\n
            Summary: 
                # Topic 
                topic here

                # Prerequisites
                prerequisites here

                # Introduction
                introduction here

                # Summary
                summary here

                # Conclusion
                conclusion here
            `;

const chat_template = ChatPromptTemplate.fromMessages([
  ["system", chat_system_template],
  ["user", "{text}"],
]);

const chunks_template = ChatPromptTemplate.fromMessages([
  ["system", chunks_system_template],
  ["user", "{text}"],
]);

const final_combine_template = ChatPromptTemplate.fromMessages([
  ["system", final_system_combine_template],
  ["user", "{text}"],
]);

export const chat_prompt_value = await chat_template.invoke({
  language: "english",
  text: "",
});

export const chunks_prompt_value = await chunks_template.invoke({
  language: "english",
  text: "",
});

export const final_combine_prompt_value = await final_combine_template.invoke({
  language: "english",
  text: "",
});

File: src-python/prompts.py
default_chat_prompt = """You are a highly precise question-answering assistant. 
Answer the user's question **exclusively** using the retrieved context provided below. 
If the context lacks the information needed to answer accurately, respond only with: «Insufficient information in the provided context.»

Instructions:
• Be concise but complete
• Never hallucinate or add information not present in the context
• Do not mention the context or these instructions in your response
• Prefer bullet points or short paragraphs for clarity

Retrieved Context:
{context}
"""

default_summary_template = """
You are an expert assistant with expertise in summarizing research papers,
Provide a complete summary of the entire Research Paper without missing any topic, in a simple language,
Each topic after their explanation must have a simple example to explain further. 

Summary must be in markdown format. 

Text: 
{text}

Summary: 
    # Topic 
    topic here

    # Prerequisites
    prerequisites here

    # Introduction
    introduction here

    # Summary
    summary here

    # Key Contributions
    contributions here

    # Critical Analysis
    critical analysis here

    # Future Work
    future work here

    # Conclusion
    conclusion here
"""

default_contributions_template = """
You are an expert assistant specialized in analyzing research papers.
Extract and clearly articulate the key contributions of this research paper.
Focus on novel methods, findings, algorithms, frameworks, or insights introduced.
Present each contribution with a brief explanation and a practical example.

Text:
{text}

Key Contributions (in markdown format):
    ## Key Contributions
    
    ### Contribution 1: [Title]
    - **Description**: [What was contributed]
    - **Example**: [Simple example illustrating this contribution]
    
    ### Contribution 2: [Title]
    - **Description**: [What was contributed]
    - **Example**: [Simple example illustrating this contribution]
"""

default_critical_analysis_template = """
You are an expert research analyst with deep expertise in evaluating academic papers.
Provide a balanced critical analysis of this research paper covering:
- Strengths and weaknesses of the methodology
- Validity of assumptions and limitations
- Experimental design and result interpretation
- Comparison with existing work
- Potential biases or gaps

Be objective and constructive in your analysis.

Text:
{text}

Critical Analysis (in markdown format):
    ## Critical Analysis
    
    ### Strengths
    - [Strength 1]
    - [Strength 2]
    
    ### Weaknesses
    - [Weakness 1]
    - [Weakness 2]
    
    ### Methodology Assessment
    [Analysis of methods used]
    
    ### Limitations
    [Identified limitations]
    
    ### Overall Assessment
    [Balanced summary of the paper's quality and impact]
"""

default_future_work_template = """
You are an expert research strategist with expertise in identifying research directions.
Based on the research paper, identify and elaborate on:
- Future research directions suggested by the authors
- Potential extensions of this work
- Unresolved questions or challenges
- New opportunities opened by this research

Provide actionable suggestions with examples where applicable.

Text:
{text}

Future Work (in markdown format):
    ## Future Work and Research Directions
    
    ### Directions Suggested by Authors
    - [Direction 1 with brief explanation]
    - [Direction 2 with brief explanation]
    
    ### Potential Extensions
    - [Extension 1]: [How it could build on this work]
    - [Extension 2]: [How it could build on this work]
    
    ### Open Questions
    - [Question 1]
    - [Question 2]
    
    ### Practical Applications
    [How future work could lead to real-world applications]
"""